---
params:
  genome: '`r params$genome`'
  dynamictitle: "Report"
  annot: '`r params$annot`'
title: "`r params$dynamictitle`"
output:
  html_document:
    toc: true
    toc_float:
      collapsed: true
      smooth_scroll: true
    css: /Users/faraz/bin/macpro/rmd.style.css
df_print: kable
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
---

<div>
<a href="https://www.rnaseqcore.vet.cornell.edu/" target="_blank" class="trex-logo"> <img src="myBin/assets/trex-extended-logo.png"
style=
"position:absolute;
top:0px;
right:0;
padding:20px;
height:140px;
text-align:justify;
display:block;
width:32%;
height:auto;
margin:0 auto 20px !important;"> </a>
</div>


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

```

## R Markdown

This report aims to provide a brief summary of methods and data diagnostic plots generated for your experiment. <br>
The fastq files are available for download upon request.

```{bash preProcess.sh, echo=F, eval=FALSE}
/home/RSCshare/share/SethupathyLab/1029_Sethupathy_FLC_2
```

Users receiving files from RSCshare are advised to delete all files once they have securely copied them to their own drives.

<font size="1"> a copy of your data will be securely archived on our end </font>

***

## sample tracking
```{r echo=F, eval = T, warning=F}
tf <- read.delim("targetFile.txt")

library(DT)
datatable(tf)
```

## Methods
The multiqc html (separate file), summarises the alignment statistics along with the duplication rates, flagstats and other useful quality control metrics pertaining to all samples.
Alignment and qc stats were generated via [bwa-mem](http://bio-bwa.sourceforge.net/bwa.shtml) and [samtools](http://www.htslib.org/doc/samtools.html);

The raw fastq reads were first processed with [trim-galore](https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md) package to:


- Trim for low quality reads;
- 2 color chemistry bias (next-seq);
- Trim for noisy short fragments;
- Trim for adapter sequence;

```{bash echo=TRUE, eval=FALSE}
trim_galore --nextseq 20 --gzip --length 50  -O 1 -a AGATCGGAAGAGC j 1 -e 0.1 --paired --fastqc
```

The trimmed reads were then aligned to `r params$genome ` reference genome with `r params$annot` annotations;

```{bash echo=TRUE, eval=FALSE}
bwa mem \
-R "@RG\tID:${name}\tSM:${name}\tPL:ILLUMINA\tLB:${name}\tPU:1" \
${genome} $i | samtools view -@ 24 -b -h -F 0x0100 -O BAM -o ${name}.bam
```

These primary bam files were then marked for duplicates using [picard](https://broadinstitute.github.io/picard/).

Duplicated and MT mapping reads were removed. The filtered (-Duplicated -MT) bams were then used for all downstream analysis.

[MACS2](https://github.com/taoliu/MACS) was used to call peaks on filtered bam files using the following parameters:

```{bash echo=TRUE, eval=FALSE}
macs2 callpeak -t ${bam} \
        -f BAMPE \
        -n ${name} \
        -g ${genomeSize} \
        -q 0.05 \
        --nomodel --shift 37 --ext 73 \
        --fe-cutoff 5 \
        --keep-dup all
```

***



## fragmentsDistribution

![](figures/countsBoxplots.png){width=1200px}

Post-normalization, the medians should be consistent across samples and more similar between biological replicates.

## sampleCluster
![](figures/cluster.png){width=800px}

An euclidean distance is computed between samples, and the dendrogram is built upon the Ward criterion. We expect this dendrogram to group replicates and separate biological conditions.



## principalComponents
![](figures/PCA.png){width=1200px}

<!-- <details> <summary> PCA without Outlier(s) </summary>

![](figures/PCA-1.png){width=1200px}
![](figures/PCA-2.png){width=1200px}


</details> -->

Another way of visualizing the experiment variability is to look at the first principal components of the PCA. On this figure, the first principal component (PC1) is expected to separate samples from the different biological conditions, meaning that the biological variability is the main source of variance in the data.

***

## citations

```{r echo=TRUE, eval=T}
citation("DESeq2")
citation("SARTools")
```
